src/controller/CHIPDeviceController.h

```diff
@@ -834,6 +858,20 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController,
                                          /* fireAndForget = */ true);
     }
 
+    void RegisterNetworkRecoverDelegate(NetworkRecoverDelegate * delegate) {
+        mNetworkRecover.SetNetworkRecoverDelegate(delegate);
+    }
+
+    /**
+     * @brief
+     *   Discover all devices advertising as recoverable.
+     *   Should be called on main loop thread.
```

Comment:
We don't generally prescribe anything about threads on which things should be called.  The rule is that you have to use whatever synchronization you use for the rest of the Matter API (which may be locks, using a single thread, or some other things).

I know you copied this comment from elsewhere in the file, but we should just fix those bits instead of copying them.